### PR TITLE
docs: add task page for setting up metamonitoring

### PR DESCRIPTION
**Pull request by tpaschalis**
This PR adds a new Task page for configuring a meta-monitoring pipeline for the agent's internal metrics, logs and traces data.

I tried to keep the page as self-contained as possible, but didn't get to the same level as other task pages to (eg. `collect-opentelemetry-data.md`).

I expect this will need a round or two of polish, but that at least it can serve as a starting point.

### CODE REVIEW

File: docs/sources/flow/tasks/metamonitoring.md [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+- /docs/grafana-cloud/agent/flow/tasks/setup-metamonitoring/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/tasks/setup-metamonitoring/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/tasks/setup-metamonitoring/
+- /docs/grafana-cloud/send-data/agent/flow/tasks/setup-metamonitoring/
+# Previous page aliases for backwards compatibility:
+- /docs/grafana-cloud/agent/flow/getting-started/setup-metamonitoring/
+- /docs/grafana-cloud/monitor-infrastructure/agent/flow/getting-started/setup-metamonitoring/
+- /docs/grafana-cloud/monitor-infrastructure/integrations/agent/flow/getting-started/setup-metamonitoring/
+- /docs/grafana-cloud/send-data/agent/flow/getting-started/setup-metamonitoring/
+- ../getting-started/setup-metamonitoring/ # /docs/agent/latest/flow/getting-started/setup-metamonitoring/
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:12]: We can remove ALL of these aliases. These are only required if we move/rename a topic that has already been published. This is a new topic... so no aliases required.

File: docs/sources/flow/tasks/metamonitoring.md [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+canonical: https://grafana.com/docs/agent/latest/flow/tasks/setup-metamonitoring/
+description: Learn how to set up meta-monitoring for Grafana Agent Flow
+title: Set up meta-monitoring
+weight: 200
+---
+
+# Set up meta-monitoring
+
+{{< param "PRODUCT_NAME" >}} can be configured to collect its own telemetry and forward it to the backend of your choosing.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:21]:
```suggestion
You can configure {{< param "PRODUCT_NAME" >}} to collect its own telemetry and forward it to the backend of your choosing.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+This topic describes how to:
+
+* Collect and forward the Agent's own metrics, logs and traces data.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:25]:
```suggestion
This topic describes how to collect and forward {{< param "PRODUCT_NAME" >}}'s metrics, logs and traces data.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+   ```river
+   prometheus.scrape "<SCRAPE_LABEL>" {
+     targets    = prometheus.exporter.<SELF_LABEL>.default.targets
+     forward_to = [<METRICS_RECEIVER_LIST>]
+   }
+   ```
+
+   Replace the following:
+   - _`<SELF_LABEL>`_: The label for the component such as `default` or `metamonitoring`. The label must be unique across all `prometheus.exporter.self` components in the same configuration file.
+   - _`<SCRAPE_LABEL>`_: The label for the scrape component such as `default`. Again, the label you use must be unique across all `prometheus.scrape` components in the same configuration file.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:62]:
```suggestion
   - _`<SCRAPE_LABEL>`_: The label for the scrape component such as `default`. The label must be unique across all `prometheus.scrape` components in the same configuration file.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+   - _`<METRICS_RECEIVER_LIST>`_: A comma-delimited list of component receivers to forward metrics to.
+     For example, to send to an existing remote write component use `prometheus.remote_write.WRITE_LABEL.receiver`. Similarly, to send data to an existing relabeling component, use `prometheus.relabel.PROCESS_LABEL.receiver`.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:64]:
```suggestion
     For example, to send to an existing remote write component, use `prometheus.remote_write.WRITE_LABEL.receiver`.
     Similarly, to send data to an existing relabeling component, use `prometheus.relabel.PROCESS_LABEL.receiver`.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+    url = "http://mimir:9009/api/v1/push"
+  }
+}
+```
+
+## Meta-monitoring logs
+
+The [logging][] block defines the logging behavior of {{< param "PRODUCT_NAME" >}}.
+
+In this task you will use the [logging][] block to forward {{< param "PRODUCT_NAME" >}}'s own logs to a compatible component. The block is specified without a label and can only be provided once per configuration file.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:89]:
```suggestion
In this task, you will use the [logging][] block to forward {{< param "PRODUCT_NAME" >}}'s logs to a compatible component. The block is specified without a label and can only be provided once per configuration file.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 7]
```diff
@@ -0,0 +1,166 @@
...
+
+1. Add the following `logging` configuration block to the top-level of your configuration file.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:91]:
```suggestion
1. Add the following `logging` configuration block to the top level of your configuration file.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 8]
```diff
@@ -0,0 +1,166 @@
...
+   logging {
+     level    = "<LOG_LEVEL>"
+     format   = "<LOG_FORMAT>"
+     write_to = [<LOGS_RECEIVER_LIST>]
+   }
+   ```
+
+   Replace the following:
+   - _`<LOG_LEVEL>`_: The log level to use for {{< param "PRODUCT_NAME" >}}'s own logs. If the attribute is not set, it defaults to `info`.
+   - _`<LOG_FORMAT>`_: The log format to use for {{< param "PRODUCT_NAME" >}}'s own logs. If the attribute is not set, it defaults to `logfmt`.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:103]:
```suggestion
   - _`<LOG_LEVEL>`_: The log level to use for {{< param "PRODUCT_NAME" >}}'s own logs. If the attribute isn't set, it defaults to `info`.
   - _`<LOG_FORMAT>`_: The log format to use for {{< param "PRODUCT_NAME" >}}'s logs. If the attribute isn't set, it defaults to `logfmt`.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 9]
```diff
@@ -0,0 +1,166 @@
...
+   - _`<LOGS_RECEIVER_LIST>`_: A comma-delimited list of component receivers to forward logs to.
+     For example, to send to an existing processing component use `loki.process.PROCESS_LABEL.receiver`. Similarly, to send data to an existing relabeling component, use `loki.relabel.PROCESS_LABEL.receiver`.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:105]:
```suggestion
     For example, to send to an existing processing component use `loki.process.PROCESS_LABEL.receiver`.
     Similarly, to send data to an existing relabeling component, use `loki.relabel.PROCESS_LABEL.receiver`.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 10]
```diff
@@ -0,0 +1,166 @@
...
+     To use data in the OTLP format, you can send data to an existing converter component like `otelcol.receiver.loki.OTEL.receiver`.
+
+The following example demonstrates configuring the config block and sending to a compatible component.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:108]:
```suggestion
The following example demonstrates configuring the configuration block and sending to a compatible component.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 11]
```diff
@@ -0,0 +1,166 @@
...
+     write_to          = [<TRACES_RECEIVER_LIST>]
+   }
+   ```
+
+   Replace the following:
+   - _`<SAMPLING_FRACTION>`_: The fraction of traces to keep. If the attribute is not set, it defaults to `0.1`.
+   - _`<TRACES_RECEIVER_LIST>`_: A comma-delimited list of component receivers to forward traces to.
+     For example, to send to an existing OpenTelemetry exporter component use `otelcol.exporter.otlp.EXPORT_LABEL.input`.
+
+The following example demonstrates configuring the config block and sending to a compatible component.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:145]:
```suggestion
The following example demonstrates configuring the configuration block and sending to a compatible component.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 12]
```diff
@@ -0,0 +1,166 @@
...
+    }
+}
+
+```
+
+## Meta-monitoring traces
+
+The [tracing][] block defines the tracing behavior of {{< param "PRODUCT_NAME" >}}.
+
+In this task you will use the [tracing][] block to forward {{< param "PRODUCT_NAME" >}} internal traces to a compatible component. The block is specified without a label and can only be provided once per configuration file.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:129]:
```suggestion
In this task you will use the [tracing][] block to forward {{< param "PRODUCT_NAME" >}} internal traces to a compatible component.
The block is specified without a label and can only be provided once per configuration file.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 13]
```diff
@@ -0,0 +1,166 @@
...
+
+1. Add the following `tracing` configuration block to the top-level of your configuration file.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:131]:
```suggestion
1. Add the following `tracing` configuration block to the top level of your configuration file.
```

File: docs/sources/flow/tasks/metamonitoring.md [hunk 14]
```diff
@@ -0,0 +1,166 @@
...
+
+   ```river
+   tracing {
+     sampling_fraction = <SAMPLING_FRACTION>
+     write_to          = [<TRACES_RECEIVER_LIST>]
+   }
+   ```
+
+   Replace the following:
+   - _`<SAMPLING_FRACTION>`_: The fraction of traces to keep. If the attribute is not set, it defaults to `0.1`.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:141]:
```suggestion
   - _`<SAMPLING_FRACTION>`_: The fraction of traces to keep. If the attribute isn't set, it defaults to `0.1`.
```

**Review by clayton-cornell — COMMENTED** [review]
Quick review and some suggestions

File: docs/sources/flow/tasks/metamonitoring.md (hunk 1 above)
Comment (tpaschalis) [reply docs/sources/flow/tasks/metamonitoring.md:12]: Thanks, TIL!

File: docs/sources/flow/tasks/metamonitoring.md [hunk 15]
```diff
@@ -0,0 +1,156 @@
...
+   }
+   ```
+
+   Replace the following:
+   - _`<LOG_LEVEL>`_: The log level to use for {{< param "PRODUCT_NAME" >}}'s logs. If the attribute isn't set, it defaults to `info`.
+   - _`<LOG_FORMAT>`_: The log format to use for {{< param "PRODUCT_NAME" >}}'s logs. If the attribute isn't set, it defaults to `logfmt`.
+   - _`<LOGS_RECEIVER_LIST>`_: A comma-delimited list of component receivers to forward logs to.
+     For example, to send to an existing processing component, use `loki.process.PROCESS_LABEL.receiver`.
+     Similarly, to send data to an existing relabeling component, use `loki.relabel.PROCESS_LABEL.receiver`.
+     To use data in the OTLP format, you can send data to an existing converter component like `otelcol.receiver.loki.OTEL.receiver`.
```
Comment (clayton-cornell) [inline docs/sources/flow/tasks/metamonitoring.md:96]:
```suggestion
     To use data in the OTLP format, you can send data to an existing converter component, like `otelcol.receiver.loki.OTEL.receiver`.
```

**Review by clayton-cornell — APPROVED** [review]
In really minor tweak Read OK otherwise to me

**Pull request merged**